src/datalakebundle/table/schema/tests/SchemaChecker_test.py

```diff
@@ -0,0 +1,378 @@
+                        t.StructField("NESTED_FIELD0", t.StringType()),
+                    ],
+                ),
+            ),
+        ),
+    ]
+)
+
+df_schema_array = t.StructType(
+    [
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+        t.StructField(
+            "ARRAY1",
+            t.ArrayType(t.StringType()),
+        ),
+    ],
+)
+
+expected_schema_array = t.StructType(
+    [
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+        t.StructField(
+            "ARRAY1",
+            t.ArrayType(t.ArrayType(t.StringType())),
+        ),
+    ],
+)
+
+df_schema_array2 = t.StructType(
+    [
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+        t.StructField(
+            "ARRAY1",
+            t.ArrayType(
+                t.ArrayType(
+                    t.StructType(
+                        [
+                            t.StructField("FIELD4", t.IntegerType()),
+                            t.StructField("FIELD5", t.IntegerType()),
+                        ]
+                    )
+                )
+            ),
+        ),
+    ],
+)
+
+expected_schema_array2 = t.StructType(
+    [
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+        t.StructField(
+            "ARRAY1",
+            t.ArrayType(t.ArrayType(t.StringType())),
+        ),
+    ],
+)
+
+df_schema_array3 = t.StructType(
+    [
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+        t.StructField(
+            "ARRAY1",
+            t.ArrayType(
+                t.StructType(
+                    [
+                        t.StructField("FIELD4", t.IntegerType()),
+                        t.StructField("FIELD5", t.IntegerType()),
+                    ]
+                )
+            ),
+        ),
+    ],
+)
+
+df_schema_array_reordered_changed_case = t.StructType(
+    [
+        t.StructField(
+            "arRay1",
+            t.ArrayType(
+                t.ArrayType(
+                    t.StructType(
+                        [
+                            t.StructField("FIELD5", t.IntegerType()),
+                            t.StructField("field4", t.IntegerType()),
+                        ]
+                    )
+                )
+            ),
+        ),
+        t.StructField("FIELD1", t.IntegerType()),
+        t.StructField("FIELD2", t.DoubleType()),
+    ],
+)
+
+
+class MyTestCase(unittest.TestCase):
+    def setUp(self) -> None:
+        self._container = bootstrapped_container.init("test")
+        logger = Logger("test")
+        mc: MetadataChecker = self._container.get(MetadataChecker)
+        self.schema_checker = SchemaChecker(logger, mc)
+
+    def test_no_diff(self):
+        assert self.schema_checker.generate_diff(df_schema, df_schema) == []
```

Comment:
self.assertEmptyArray() or any other self.assert* method should be used.

Reply:
self.assertEmptyArray() doesn't exist so I used self.assertFalse() instead
Fixed the rest

Reply:
sorry, there are not arrays in python, my PHP bad :-)